From 722a54b38231218e5091a8fa7c138f582066c5a6 Mon Sep 17 00:00:00 2001
From: Haibo Chen <haibo.chen@nxp.com>
Date: Fri, 9 Feb 2018 18:02:18 +0800
Subject: [PATCH 3480/5242] MLK-17579 ARM64: dts: correct the pad setting for
 imx8 usdhc

commit  595906e2846079b24ab1c3de3e8ba77c656ff438 from
https://source.codeaurora.org/external/imx/linux-imx.git

according to IC suggestion, usdhc clock pad need to be configed as
input/output mode, for other usdhc pad, including the strobe pad, need
to be configed as normal mode.

This patch do the change on the imx8qxp and imx8qm board.

Reviewed-by: Dong Aisheng <aisheng.dong@nxp.com>
Signed-off-by: Haibo Chen <haibo.chen@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../boot/dts/freescale/fsl-imx8qm-lpddr4-arm2.dts  |    6 +--
 arch/arm64/boot/dts/freescale/fsl-imx8qm-mek.dts   |    6 +--
 .../boot/dts/freescale/fsl-imx8qxp-lpddr4-arm2.dts |   48 ++++++++++----------
 arch/arm64/boot/dts/freescale/fsl-imx8qxp-mek.dts  |   48 ++++++++++----------
 4 files changed, 54 insertions(+), 54 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qm-lpddr4-arm2.dts b/arch/arm64/boot/dts/freescale/fsl-imx8qm-lpddr4-arm2.dts
index 13dd2dbc..e08349d7 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qm-lpddr4-arm2.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qm-lpddr4-arm2.dts
@@ -362,7 +362,7 @@
 				SC_P_EMMC0_DATA5_CONN_EMMC0_DATA5	0x00000021
 				SC_P_EMMC0_DATA6_CONN_EMMC0_DATA6	0x00000021
 				SC_P_EMMC0_DATA7_CONN_EMMC0_DATA7	0x00000021
-				SC_P_EMMC0_STROBE_CONN_EMMC0_STROBE	0x06000041
+				SC_P_EMMC0_STROBE_CONN_EMMC0_STROBE	0x00000041
 				SC_P_EMMC0_RESET_B_CONN_EMMC0_RESET_B	0x00000021
 			>;
 		};
@@ -379,7 +379,7 @@
 				SC_P_EMMC0_DATA5_CONN_EMMC0_DATA5	0x00000020
 				SC_P_EMMC0_DATA6_CONN_EMMC0_DATA6	0x00000020
 				SC_P_EMMC0_DATA7_CONN_EMMC0_DATA7	0x00000020
-				SC_P_EMMC0_STROBE_CONN_EMMC0_STROBE	0x06000040
+				SC_P_EMMC0_STROBE_CONN_EMMC0_STROBE	0x00000040
 				SC_P_EMMC0_RESET_B_CONN_EMMC0_RESET_B	0x00000020
 			>;
 		};
@@ -396,7 +396,7 @@
 				SC_P_EMMC0_DATA5_CONN_EMMC0_DATA5	0x00000020
 				SC_P_EMMC0_DATA6_CONN_EMMC0_DATA6	0x00000020
 				SC_P_EMMC0_DATA7_CONN_EMMC0_DATA7	0x00000020
-				SC_P_EMMC0_STROBE_CONN_EMMC0_STROBE	0x06000040
+				SC_P_EMMC0_STROBE_CONN_EMMC0_STROBE	0x00000040
 				SC_P_EMMC0_RESET_B_CONN_EMMC0_RESET_B	0x00000020
 			>;
 		};
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek.dts b/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek.dts
index caaa298..7539b5b 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek.dts
@@ -404,7 +404,7 @@
 				SC_P_EMMC0_DATA5_CONN_EMMC0_DATA5	0x00000021
 				SC_P_EMMC0_DATA6_CONN_EMMC0_DATA6	0x00000021
 				SC_P_EMMC0_DATA7_CONN_EMMC0_DATA7	0x00000021
-				SC_P_EMMC0_STROBE_CONN_EMMC0_STROBE	0x06000041
+				SC_P_EMMC0_STROBE_CONN_EMMC0_STROBE	0x00000041
 				SC_P_EMMC0_RESET_B_CONN_EMMC0_RESET_B	0x00000021
 			>;
 		};
@@ -421,7 +421,7 @@
 				SC_P_EMMC0_DATA5_CONN_EMMC0_DATA5	0x00000020
 				SC_P_EMMC0_DATA6_CONN_EMMC0_DATA6	0x00000020
 				SC_P_EMMC0_DATA7_CONN_EMMC0_DATA7	0x00000020
-				SC_P_EMMC0_STROBE_CONN_EMMC0_STROBE	0x06000040
+				SC_P_EMMC0_STROBE_CONN_EMMC0_STROBE	0x00000040
 				SC_P_EMMC0_RESET_B_CONN_EMMC0_RESET_B	0x00000020
 			>;
 		};
@@ -438,7 +438,7 @@
 				SC_P_EMMC0_DATA5_CONN_EMMC0_DATA5	0x00000020
 				SC_P_EMMC0_DATA6_CONN_EMMC0_DATA6	0x00000020
 				SC_P_EMMC0_DATA7_CONN_EMMC0_DATA7	0x00000020
-				SC_P_EMMC0_STROBE_CONN_EMMC0_STROBE	0x06000040
+				SC_P_EMMC0_STROBE_CONN_EMMC0_STROBE	0x00000040
 				SC_P_EMMC0_RESET_B_CONN_EMMC0_RESET_B	0x00000020
 			>;
 		};
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qxp-lpddr4-arm2.dts b/arch/arm64/boot/dts/freescale/fsl-imx8qxp-lpddr4-arm2.dts
index 9ff6f27..93c2bc5 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qxp-lpddr4-arm2.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qxp-lpddr4-arm2.dts
@@ -293,7 +293,7 @@
 				SC_P_EMMC0_DATA5_CONN_EMMC0_DATA5	0x00000021
 				SC_P_EMMC0_DATA6_CONN_EMMC0_DATA6	0x00000021
 				SC_P_EMMC0_DATA7_CONN_EMMC0_DATA7	0x00000021
-				SC_P_EMMC0_STROBE_CONN_EMMC0_STROBE	0x06000041
+				SC_P_EMMC0_STROBE_CONN_EMMC0_STROBE	0x00000041
 				SC_P_EMMC0_RESET_B_CONN_EMMC0_RESET_B	0x00000021
 			>;
 		};
@@ -310,7 +310,7 @@
 				SC_P_EMMC0_DATA5_CONN_EMMC0_DATA5	0x00000020
 				SC_P_EMMC0_DATA6_CONN_EMMC0_DATA6	0x00000020
 				SC_P_EMMC0_DATA7_CONN_EMMC0_DATA7	0x00000020
-				SC_P_EMMC0_STROBE_CONN_EMMC0_STROBE	0x06000040
+				SC_P_EMMC0_STROBE_CONN_EMMC0_STROBE	0x00000040
 				SC_P_EMMC0_RESET_B_CONN_EMMC0_RESET_B	0x00000020
 			>;
 		};
@@ -327,52 +327,52 @@
 				SC_P_EMMC0_DATA5_CONN_EMMC0_DATA5	0x00000020
 				SC_P_EMMC0_DATA6_CONN_EMMC0_DATA6	0x00000020
 				SC_P_EMMC0_DATA7_CONN_EMMC0_DATA7	0x00000020
-				SC_P_EMMC0_STROBE_CONN_EMMC0_STROBE	0x06000040
+				SC_P_EMMC0_STROBE_CONN_EMMC0_STROBE	0x00000040
 				SC_P_EMMC0_RESET_B_CONN_EMMC0_RESET_B	0x00000020
 			>;
 		};
 
 		pinctrl_usdhc2_gpio: usdhc2gpiogrp {
 			fsl,pins = <
-				SC_P_USDHC1_RESET_B_LSIO_GPIO4_IO19	0x06000021
-				SC_P_USDHC1_WP_LSIO_GPIO4_IO21		0x06000021
-				SC_P_USDHC1_CD_B_LSIO_GPIO4_IO22	0x06000021
+				SC_P_USDHC1_RESET_B_LSIO_GPIO4_IO19	0x00000021
+				SC_P_USDHC1_WP_LSIO_GPIO4_IO21		0x00000021
+				SC_P_USDHC1_CD_B_LSIO_GPIO4_IO22	0x00000021
 			>;
 		};
 
 		pinctrl_usdhc2: usdhc2grp {
 			fsl,pins = <
 				SC_P_USDHC1_CLK_CONN_USDHC1_CLK		0x06000041
-				SC_P_USDHC1_CMD_CONN_USDHC1_CMD		0x06000021
-				SC_P_USDHC1_DATA0_CONN_USDHC1_DATA0	0x06000021
-				SC_P_USDHC1_DATA1_CONN_USDHC1_DATA1	0x06000021
-				SC_P_USDHC1_DATA2_CONN_USDHC1_DATA2	0x06000021
-				SC_P_USDHC1_DATA3_CONN_USDHC1_DATA3	0x06000021
-				SC_P_USDHC1_VSELECT_CONN_USDHC1_VSELECT	0x06000021
+				SC_P_USDHC1_CMD_CONN_USDHC1_CMD		0x00000021
+				SC_P_USDHC1_DATA0_CONN_USDHC1_DATA0	0x00000021
+				SC_P_USDHC1_DATA1_CONN_USDHC1_DATA1	0x00000021
+				SC_P_USDHC1_DATA2_CONN_USDHC1_DATA2	0x00000021
+				SC_P_USDHC1_DATA3_CONN_USDHC1_DATA3	0x00000021
+				SC_P_USDHC1_VSELECT_CONN_USDHC1_VSELECT	0x00000021
 			>;
 		};
 
 		pinctrl_usdhc2_100mhz: usdhc2grp100mhz {
 			fsl,pins = <
 				SC_P_USDHC1_CLK_CONN_USDHC1_CLK		0x06000040
-				SC_P_USDHC1_CMD_CONN_USDHC1_CMD		0x06000020
-				SC_P_USDHC1_DATA0_CONN_USDHC1_DATA0	0x06000020
-				SC_P_USDHC1_DATA1_CONN_USDHC1_DATA1	0x06000020
-				SC_P_USDHC1_DATA2_CONN_USDHC1_DATA2	0x06000020
-				SC_P_USDHC1_DATA3_CONN_USDHC1_DATA3	0x06000020
-				SC_P_USDHC1_VSELECT_CONN_USDHC1_VSELECT	0x06000020
+				SC_P_USDHC1_CMD_CONN_USDHC1_CMD		0x00000020
+				SC_P_USDHC1_DATA0_CONN_USDHC1_DATA0	0x00000020
+				SC_P_USDHC1_DATA1_CONN_USDHC1_DATA1	0x00000020
+				SC_P_USDHC1_DATA2_CONN_USDHC1_DATA2	0x00000020
+				SC_P_USDHC1_DATA3_CONN_USDHC1_DATA3	0x00000020
+				SC_P_USDHC1_VSELECT_CONN_USDHC1_VSELECT	0x00000020
 			>;
 		};
 
 		pinctrl_usdhc2_200mhz: usdhc2grp200mhz {
 			fsl,pins = <
 				SC_P_USDHC1_CLK_CONN_USDHC1_CLK		0x06000040
-				SC_P_USDHC1_CMD_CONN_USDHC1_CMD		0x06000020
-				SC_P_USDHC1_DATA0_CONN_USDHC1_DATA0	0x06000020
-				SC_P_USDHC1_DATA1_CONN_USDHC1_DATA1	0x06000020
-				SC_P_USDHC1_DATA2_CONN_USDHC1_DATA2	0x06000020
-				SC_P_USDHC1_DATA3_CONN_USDHC1_DATA3	0x06000020
-				SC_P_USDHC1_VSELECT_CONN_USDHC1_VSELECT	0x06000020
+				SC_P_USDHC1_CMD_CONN_USDHC1_CMD		0x00000020
+				SC_P_USDHC1_DATA0_CONN_USDHC1_DATA0	0x00000020
+				SC_P_USDHC1_DATA1_CONN_USDHC1_DATA1	0x00000020
+				SC_P_USDHC1_DATA2_CONN_USDHC1_DATA2	0x00000020
+				SC_P_USDHC1_DATA3_CONN_USDHC1_DATA3	0x00000020
+				SC_P_USDHC1_VSELECT_CONN_USDHC1_VSELECT	0x00000020
 			>;
 		};
 
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qxp-mek.dts b/arch/arm64/boot/dts/freescale/fsl-imx8qxp-mek.dts
index c771db6..8181103 100755
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qxp-mek.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qxp-mek.dts
@@ -354,7 +354,7 @@
 				SC_P_EMMC0_DATA5_CONN_EMMC0_DATA5	0x00000021
 				SC_P_EMMC0_DATA6_CONN_EMMC0_DATA6	0x00000021
 				SC_P_EMMC0_DATA7_CONN_EMMC0_DATA7	0x00000021
-				SC_P_EMMC0_STROBE_CONN_EMMC0_STROBE	0x06000041
+				SC_P_EMMC0_STROBE_CONN_EMMC0_STROBE	0x00000041
 			>;
 		};
 
@@ -377,7 +377,7 @@
 				SC_P_EMMC0_DATA5_CONN_EMMC0_DATA5	0x00000020
 				SC_P_EMMC0_DATA6_CONN_EMMC0_DATA6	0x00000020
 				SC_P_EMMC0_DATA7_CONN_EMMC0_DATA7	0x00000020
-				SC_P_EMMC0_STROBE_CONN_EMMC0_STROBE	0x06000040
+				SC_P_EMMC0_STROBE_CONN_EMMC0_STROBE	0x00000040
 			>;
 		};
 
@@ -393,51 +393,51 @@
 				SC_P_EMMC0_DATA5_CONN_EMMC0_DATA5	0x00000020
 				SC_P_EMMC0_DATA6_CONN_EMMC0_DATA6	0x00000020
 				SC_P_EMMC0_DATA7_CONN_EMMC0_DATA7	0x00000020
-				SC_P_EMMC0_STROBE_CONN_EMMC0_STROBE	0x06000040
+				SC_P_EMMC0_STROBE_CONN_EMMC0_STROBE	0x00000040
 			>;
 		};
 
 		pinctrl_usdhc2_gpio: usdhc2gpiogrp {
 			fsl,pins = <
-				SC_P_USDHC1_RESET_B_LSIO_GPIO4_IO19	0x06000021
-				SC_P_USDHC1_WP_LSIO_GPIO4_IO21		0x06000021
-				SC_P_USDHC1_CD_B_LSIO_GPIO4_IO22	0x06000021
+				SC_P_USDHC1_RESET_B_LSIO_GPIO4_IO19	0x00000021
+				SC_P_USDHC1_WP_LSIO_GPIO4_IO21		0x00000021
+				SC_P_USDHC1_CD_B_LSIO_GPIO4_IO22	0x00000021
 			>;
 		};
 
 		pinctrl_usdhc2: usdhc2grp {
 			fsl,pins = <
 				SC_P_USDHC1_CLK_CONN_USDHC1_CLK		0x06000041
-				SC_P_USDHC1_CMD_CONN_USDHC1_CMD		0x06000021
-				SC_P_USDHC1_DATA0_CONN_USDHC1_DATA0	0x06000021
-				SC_P_USDHC1_DATA1_CONN_USDHC1_DATA1	0x06000021
-				SC_P_USDHC1_DATA2_CONN_USDHC1_DATA2	0x06000021
-				SC_P_USDHC1_DATA3_CONN_USDHC1_DATA3	0x06000021
-				SC_P_USDHC1_VSELECT_CONN_USDHC1_VSELECT	0x06000021
+				SC_P_USDHC1_CMD_CONN_USDHC1_CMD		0x00000021
+				SC_P_USDHC1_DATA0_CONN_USDHC1_DATA0	0x00000021
+				SC_P_USDHC1_DATA1_CONN_USDHC1_DATA1	0x00000021
+				SC_P_USDHC1_DATA2_CONN_USDHC1_DATA2	0x00000021
+				SC_P_USDHC1_DATA3_CONN_USDHC1_DATA3	0x00000021
+				SC_P_USDHC1_VSELECT_CONN_USDHC1_VSELECT	0x00000021
 			>;
 		};
 
 		pinctrl_usdhc2_100mhz: usdhc2grp100mhz {
 			fsl,pins = <
 				SC_P_USDHC1_CLK_CONN_USDHC1_CLK		0x06000040
-				SC_P_USDHC1_CMD_CONN_USDHC1_CMD		0x06000020
-				SC_P_USDHC1_DATA0_CONN_USDHC1_DATA0	0x06000020
-				SC_P_USDHC1_DATA1_CONN_USDHC1_DATA1	0x06000020
-				SC_P_USDHC1_DATA2_CONN_USDHC1_DATA2	0x06000020
-				SC_P_USDHC1_DATA3_CONN_USDHC1_DATA3	0x06000020
-				SC_P_USDHC1_VSELECT_CONN_USDHC1_VSELECT	0x06000020
+				SC_P_USDHC1_CMD_CONN_USDHC1_CMD		0x00000020
+				SC_P_USDHC1_DATA0_CONN_USDHC1_DATA0	0x00000020
+				SC_P_USDHC1_DATA1_CONN_USDHC1_DATA1	0x00000020
+				SC_P_USDHC1_DATA2_CONN_USDHC1_DATA2	0x00000020
+				SC_P_USDHC1_DATA3_CONN_USDHC1_DATA3	0x00000020
+				SC_P_USDHC1_VSELECT_CONN_USDHC1_VSELECT	0x00000020
 			>;
 		};
 
 		pinctrl_usdhc2_200mhz: usdhc2grp200mhz {
 			fsl,pins = <
 				SC_P_USDHC1_CLK_CONN_USDHC1_CLK		0x06000040
-				SC_P_USDHC1_CMD_CONN_USDHC1_CMD		0x06000020
-				SC_P_USDHC1_DATA0_CONN_USDHC1_DATA0	0x06000020
-				SC_P_USDHC1_DATA1_CONN_USDHC1_DATA1	0x06000020
-				SC_P_USDHC1_DATA2_CONN_USDHC1_DATA2	0x06000020
-				SC_P_USDHC1_DATA3_CONN_USDHC1_DATA3	0x06000020
-				SC_P_USDHC1_VSELECT_CONN_USDHC1_VSELECT	0x06000020
+				SC_P_USDHC1_CMD_CONN_USDHC1_CMD		0x00000020
+				SC_P_USDHC1_DATA0_CONN_USDHC1_DATA0	0x00000020
+				SC_P_USDHC1_DATA1_CONN_USDHC1_DATA1	0x00000020
+				SC_P_USDHC1_DATA2_CONN_USDHC1_DATA2	0x00000020
+				SC_P_USDHC1_DATA3_CONN_USDHC1_DATA3	0x00000020
+				SC_P_USDHC1_VSELECT_CONN_USDHC1_VSELECT	0x00000020
 			>;
 		};
 
-- 
1.7.9.5

